core/trino-main/src/main/java/io/trino/operator/exchange/PageReference.java

```diff
@@ -31,26 +31,28 @@
     private volatile int referenceCount;
     private final Page page;
     private final PageReleasedListener onPageReleased;
+    private final long retainedSizeInBytes;
 
     public PageReference(Page page, int referenceCount, PageReleasedListener onPageReleased)
     {
         this.page = requireNonNull(page, "page is null");
         this.onPageReleased = requireNonNull(onPageReleased, "onPageReleased is null");
         checkArgument(referenceCount >= 1, "referenceCount must be at least 1");
         this.referenceCount = referenceCount;
+        this.retainedSizeInBytes = page.getRetainedSizeInBytes();
```

Comment:
add a comment why it's fine to precalculate this (which is potentially expensive) even though `getRetainedSizeInBytes` wasn't called.

Reply:
Done